target_chains/solana/cli/src/main.rs

```diff
@@ -10,48 +10,57 @@ use {
     std::str::FromStr,
 
     solana_sdk::{
-        pubkey::Pubkey,
         signature::{
             read_keypair_file,
             Keypair,
         },
         signer::Signer,
+        pubkey::Pubkey,
         instruction::Instruction,
         transaction::Transaction,
     },
 
     wormhole::VAA,
     wormhole_solana::{
-        Account,
-        GuardianSet,
-        Config as WormholeConfig,
         instructions::{
             post_vaa,
             verify_signatures_txs,
             PostVAAData,
         },
+        Account,
+        GuardianSet,
+        Config as WormholeConfig,
+        VAA as WormholeSolanaVAA,
+    },
+
+    pyth_solana_receiver::{
+        ID,
+        accounts::DecodePostedVaa,
+    },
+
+    anchor_client::anchor_lang::{
+        ToAccountMetas,
+        InstructionData,
+        AnchorDeserialize,
     },
 
     solana_client::rpc_client::RpcClient,
-    anchor_client::anchor_lang::AnchorDeserialize,
 };
 
 fn main() -> Result<()> {
     let cli = Cli::parse();
 
     match cli.action {
-        Action::PostPriceVAA { vaa, keypair } => {
-            println!("PostPriceVAA is invoked with vaa\"{}\"", vaa);
-            // Hard-coded strings
+        Action::PostAndReceiveVAA { vaa, keypair } => {
             let rpc_client = RpcClient::new("https://api.devnet.solana.com");
-            // Is RpcClient::new_with_commitment necessary?
             let wormhole = Pubkey::from_str("3u8hJUVTA4jH1wYAyUur7FFZVQ8H635K3tSHHF4ssjQ5").unwrap();
```

Comment:
`let wormhole = AnchorVaa::owner();` so that we only hardcode in a single place

Reply:
Good point. Sorry for missing this.